### PR TITLE
Deprecated legacy config settings & telemetry

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://mattermost.atlassian.net/browse/MM-57106

Deprecated the following HA config settings: MaxIdleConns, MaxIdleConnsPerHost, and IdleConnTimeoutMill…iseconds. Also removed multiple deprecated settings from telemetry docs documented erroneously.

### CODE REVIEW

File: source/configure/deprecated-configuration-settings.rst [hunk 1]
```diff
@@ -382,6 +381,42 @@ The port used for streaming data between servers.
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*Deprecated. Not used in Mattermost v7.0 and later*
+
++-----------------------------------------------------------------+------------------------------------------------------------------------+
+| The maximum number of idle connections held open from one       | - System Config path: N/A                                              |
+| server to all others in the cluster.                            | - ``config.json`` setting: ``".ClusterSettings.MaxIdleConns: 100,``    |
+|                                                                 | - Environment variable: ``MM_CLUSTERSETTINGS_MAXIDLECONNS``            |
+| Numerical input. Default is **100**.                            |                                                                        |
++-----------------------------------------------------------------+------------------------------------------------------------------------+
```
Comment (hanzei) [inline source/configure/deprecated-configuration-settings.rst:394]: The formatting is different to the other deprecated settings in the file. Is that intended?

**Review by hanzei — CHANGES_REQUESTED** [review]
On question regarding the formatting in `source/configure/deprecated-configuration-settings.rst`

File: source/configure/deprecated-configuration-settings.rst (hunk 1 above)
Comment (hanzei) [reply source/configure/deprecated-configuration-settings.rst:394]: The only existing one I could find is `Enable publicly-shared boards` (https://github.com/mattermost/docs/blob/deprecate-legacy-configs/source/configure/deprecated-configuration-settings.rst#enable-publicly-shared-boards).

I'm 0/5 on what format to use. 1/5 that it looks a bit inconsistent to use different formats.

**Review by hanzei — APPROVED** [review]

**Comment by cwarnermm** [comment]
Agreed that the inconsistency isn't ideal; however, resources are limited and consistency on a deprecation page isn't a priority at this time.

**Pull request merged**